guiEditor/src/diagram/workbench.tsx

```diff
@@ -157,7 +165,9 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
 
     ctrlEvent = (evt: KeyboardEvent) => {
         this._ctrlKeyIsPressed = evt.ctrlKey;
-
+        this._constraintDirection = evt.shiftKey ?

```

Comment:
Kind of hard to decipher ??? might be good on several lines ? (or I am getting old ... or both :-) )